src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -89,6 +99,27 @@ internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, R
 
                 if (response.StatusCode != HttpStatusCode.OK || expectErrorsOn200OK)
                 {
+                    //Check if we can perform device auth
+                    if (_deviceAuthManager != null && IsDeviceAuthChallenge(response, respondToDeviceAuthChallenge))
```

Comment:
Can we have the deviceManager take over the entire responsability of looking at an HTTPResponse and returning a Header? THe OAuth client shouldn't do this. 

Smth like 

`string header = await _deviceManager.TryCreateDeviceAuthChallengeAsync(response)`